<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>502</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ESP8266 firmware flasher</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resource.qrc">
    <normaloff>:/images/res/images/app/128.png</normaloff>:/images/res/images/app/128.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>ESP device</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <property name="labelAlignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Serial port:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QComboBox" name="serialPort">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>30</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>30</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="baudRate">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>30</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>30</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="resetMode">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>30</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>30</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="openBtn">
                <property name="toolTip">
                 <string>Connect to serial port</string>
                </property>
                <property name="statusTip">
                 <string>Connect to serial port</string>
                </property>
                <property name="whatsThis">
                 <string>Connect to serial port</string>
                </property>
                <property name="styleSheet">
                 <string notr="true">text-align:left;
font-weight:bold;
background-color:rgb(62, 152, 9);</string>
                </property>
                <property name="text">
                 <string>Connect</string>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/res/images/light/appbar.usb.png</normaloff>:/images/res/images/light/appbar.usb.png</iconset>
                </property>
                <property name="iconSize">
                 <size>
                  <width>24</width>
                  <height>24</height>
                 </size>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>SPI Flash mode:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="spiMode">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>30</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>SPI Flash size:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QComboBox" name="flashSize">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>30</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>SPI Flash speed:</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QComboBox" name="spiSpeed">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>30</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QGroupBox" name="macAddressGroup">
            <property name="title">
             <string>MAC address:</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QLabel" name="macBarcodeLabel">
               <property name="font">
                <font>
                 <pointsize>20</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QPushButton" name="printMacBtn">
                 <property name="toolTip">
                  <string>Print MAC address barcode</string>
                 </property>
                 <property name="statusTip">
                  <string>Print MAC address barcode</string>
                 </property>
                 <property name="whatsThis">
                  <string>Print MAC address barcode</string>
                 </property>
                 <property name="text">
                  <string>Print</string>
                 </property>
                 <property name="icon">
                  <iconset resource="resource.qrc">
                   <normaloff>:/images/res/images/light/appbar.printer.text.png</normaloff>:/images/res/images/light/appbar.printer.text.png</iconset>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>24</width>
                   <height>24</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="copyMacBtn">
                 <property name="toolTip">
                  <string>Copy MAC address to clipboard</string>
                 </property>
                 <property name="statusTip">
                  <string>Copy MAC address to clipboard</string>
                 </property>
                 <property name="whatsThis">
                  <string>Copy MAC address to clipboard</string>
                 </property>
                 <property name="text">
                  <string>Copy</string>
                 </property>
                 <property name="icon">
                  <iconset resource="resource.qrc">
                   <normaloff>:/images/res/images/light/appbar.clipboard.variant.text.png</normaloff>:/images/res/images/light/appbar.clipboard.variant.text.png</iconset>
                 </property>
                 <property name="iconSize">
                  <size>
                   <width>24</width>
                   <height>24</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="filesTab">
       <attribute name="title">
        <string>Firmware files</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayoutFT"/>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="flashActionsGroup">
      <property name="title">
       <string>SPI Flash actions:</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="0">
          <widget class="QPushButton" name="eraseFlashBtn">
           <property name="toolTip">
            <string>Perform Chip Erase on SPI flash.</string>
           </property>
           <property name="statusTip">
            <string>Perform Chip Erase on SPI flash.</string>
           </property>
           <property name="whatsThis">
            <string>Perform Chip Erase on SPI flash.</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;
font-weight: bold;</string>
           </property>
           <property name="text">
            <string>Erase Flash</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.close.png</normaloff>:/images/res/images/light/appbar.close.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="readFlashBtn">
           <property name="toolTip">
            <string>Read SPI flash content</string>
           </property>
           <property name="statusTip">
            <string>Read SPI flash content</string>
           </property>
           <property name="whatsThis">
            <string>Read SPI flash content</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Read Flash</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.disk.download.png</normaloff>:/images/res/images/light/appbar.disk.download.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QPushButton" name="writeFlashBtn">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Write a binary blob to flash</string>
           </property>
           <property name="statusTip">
            <string>Write a binary blob to flash</string>
           </property>
           <property name="whatsThis">
            <string>Write a binary blob to flash</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;
font-weight: bold;
background-color:rgb(62, 152, 9);</string>
           </property>
           <property name="text">
            <string>Write Flash</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.disk.upload.png</normaloff>:/images/res/images/light/appbar.disk.upload.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="writeMemoryBtn">
           <property name="toolTip">
            <string>Write to arbitrary memory location</string>
           </property>
           <property name="statusTip">
            <string>Write to arbitrary memory location</string>
           </property>
           <property name="whatsThis">
            <string>Write to arbitrary memory location</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;
font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Write Memory</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.disk.upload.png</normaloff>:/images/res/images/light/appbar.disk.upload.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="loadRamBtn">
           <property name="toolTip">
            <string>Download an image to RAM and execute</string>
           </property>
           <property name="statusTip">
            <string>Download an image to RAM and execute</string>
           </property>
           <property name="whatsThis">
            <string>Download an image to RAM and execute</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Load Ram</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.disk.upload.png</normaloff>:/images/res/images/light/appbar.disk.upload.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="readMemoryBtn">
           <property name="toolTip">
            <string>Read arbitrary memory location</string>
           </property>
           <property name="statusTip">
            <string>Read arbitrary memory location</string>
           </property>
           <property name="whatsThis">
            <string>Read arbitrary memory location</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Read Memory</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.disk.download.png</normaloff>:/images/res/images/light/appbar.disk.download.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="dumpMemoryBtn">
           <property name="toolTip">
            <string>Dump arbitrary memory to disk</string>
           </property>
           <property name="statusTip">
            <string>Dump arbitrary memory to disk</string>
           </property>
           <property name="whatsThis">
            <string>Dump arbitrary memory to disk</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Dump Memory</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.disk.download.png</normaloff>:/images/res/images/light/appbar.disk.download.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="runImageBtn">
           <property name="toolTip">
            <string>Run application code in flash</string>
           </property>
           <property name="statusTip">
            <string>Run application code in flash</string>
           </property>
           <property name="whatsThis">
            <string>Run application code in flash</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Run Image</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.rocket.png</normaloff>:/images/res/images/light/appbar.rocket.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="makeImageBtn">
           <property name="toolTip">
            <string>Create an application image from binary files</string>
           </property>
           <property name="statusTip">
            <string>Create an application image from binary files</string>
           </property>
           <property name="whatsThis">
            <string>Create an application image from binary files</string>
           </property>
           <property name="styleSheet">
            <string notr="true">text-align:left;font-weight:bold;</string>
           </property>
           <property name="text">
            <string>Image Tools...</string>
           </property>
           <property name="icon">
            <iconset resource="resource.qrc">
             <normaloff>:/images/res/images/light/appbar.cog.png</normaloff>:/images/res/images/light/appbar.cog.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="LogList" name="logList">
         <property name="font">
          <font>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="showDropIndicator" stdset="0">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer_4">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionPreferences"/>
    <addaction name="actionImport_image_file_list"/>
    <addaction name="actionExport_image_file_list"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>&amp;?</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionImport_image_file_list">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/images/res/images/light/appbar.inbox.in.png</normaloff>:/images/res/images/light/appbar.inbox.in.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Import image file list...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionExport_image_file_list">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/images/res/images/light/appbar.inbox.out.png</normaloff>:/images/res/images/light/appbar.inbox.out.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Export image file list...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/images/res/images/light/appbar.question.png</normaloff>:/images/res/images/light/appbar.question.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About espflasher...</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>Preferences...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>LogList</class>
   <extends>QListWidget</extends>
   <header>loglist.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
